liquidity-mining/src/tests/tests.rs

```diff
@@ -0,0 +1,2186 @@
+            pretty_assertions::assert_eq!(yield_farm.updated_at, current_period);
+
+            //IV. - booth farms met conditions for update
+            let current_period = 30;
+            assert_ok!(LiquidityMining::maybe_update_farms(
+                &mut global_farm,
+                &mut yield_farm,
+                current_period
+            ));
+
+            assert_ne!(global_farm, expected_global_farm);
+            assert_ne!(yield_farm, expected_yield_farm);
+
+            pretty_assertions::assert_eq!(global_farm.updated_at, current_period);
+            pretty_assertions::assert_eq!(yield_farm.updated_at, current_period);
+
+            TransactionOutcome::Commit(())
+        });
+    });
+}
+
+#[test]
+fn depositdata_add_farm_entry_to_should_work() {
+    let mut deposit = DepositData::<Test, Instance1> {
+        shares: 10,
+        amm_pool_id: BSX_TKN1_AMM,
+        yield_farm_entries: vec![].try_into().unwrap(),
+    };
+
+    let test_farm_entries = vec![
+        YieldFarmEntry::<Test, Instance1>::new(1, 50, 20, FixedU128::from(12), 2),
+        YieldFarmEntry::<Test, Instance1>::new(2, 18, 20, FixedU128::from(14), 18),
+        YieldFarmEntry::<Test, Instance1>::new(3, 60, 20, FixedU128::from(1), 1),
+        YieldFarmEntry::<Test, Instance1>::new(4, 1, 20, FixedU128::from(10), 13),
+        YieldFarmEntry::<Test, Instance1>::new(7, 2, 20, FixedU128::from(10), 13),
+        YieldFarmEntry::<Test, Instance1>::new(5, 100, 20, FixedU128::from(10), 13),
+        YieldFarmEntry::<Test, Instance1>::new(6, 4, 20, FixedU128::from(10), 13),
+    ];
+
+    assert_ok!(deposit.add_yield_farm_entry(test_farm_entries[0].clone()));
+
+    assert_ok!(deposit.add_yield_farm_entry(test_farm_entries[2].clone()));
+
+    assert_ok!(deposit.add_yield_farm_entry(test_farm_entries[3].clone()));
+
+    //`yield_farm_id` must be unique in `yield_farm_entries`
+    assert_err!(
+        deposit.add_yield_farm_entry(test_farm_entries[2].clone()),
+        Error::<Test, Instance1>::DoubleLock
+    );
+    assert_err!(
+        deposit.add_yield_farm_entry(YieldFarmEntry::<Test, Instance1>::new(1, 50, 10, FixedU128::from(1), 1)),
+        Error::<Test, Instance1>::DoubleLock
+    );
+
+    assert_ok!(deposit.add_yield_farm_entry(test_farm_entries[4].clone()));
+
+    assert_ok!(deposit.add_yield_farm_entry(test_farm_entries[6].clone()));
+
+    pretty_assertions::assert_eq!(
+        deposit,
+        DepositData::<Test, Instance1> {
+            shares: 10,
+            amm_pool_id: BSX_TKN1_AMM,
+            yield_farm_entries: vec![
+                test_farm_entries[0].clone(),
+                test_farm_entries[2].clone(),
+                test_farm_entries[3].clone(),
+                test_farm_entries[4].clone(),
+                test_farm_entries[6].clone(),
+            ]
+            .try_into()
+            .unwrap(),
+        }
+    );
+
+    //5 is max farm entries.
+    assert_err!(
```

Comment:
should probably be `assert_noop`
```suggestion
    assert_noop!(
```
same for the other `assert_err`s

Reply:
`assert_err` is used when fns doesn't needs externalities. This are failable fns without storage operations.

Reply:
Ah, that makes sense, sorry I missed it.